### ISSUE TITLE
Simple program with an array cannot be verified

### ISSUE THREAD

**Issue by pruemmer** [jayhorn/jayhorn#68]
With options `-rta -mem-prec 0`, the following program cannot be verified:

```
import java.util.Random;

public class Test {
    public static void main(String[] args) {
        if (args.length < 1)
            return;

        final int N = Integer.parseInt(args[0]);
        Random randomGenerator = new Random();

        if (N < 10 || N > 100)
            return;

        int data[] = new int[N];
        for (int i = 0; i < N; ++i)
            data[i] = i;
    }
}
```

something seems to be wrong with the way array length is handled here?

I get result SAFE if the option `-mem-prec 0` is removed, but the solution looks too simple.

**Comment by martinschaef** [comment]
Yes, something is fishy. Added [this test case](https://github.com/jayhorn/jayhorn/blob/bcf403ea5e8e12537186d664a3fd2c9248a81908/jayhorn/src/test/resources/horn-encoding/backlog/UnsatIssue68.java). It passes with `-mem-prec 0` and fails otherwise. @rodykersten that's your ball :)

**Comment by martinschaef** [comment]
Not sure why this is closed. For me it still throws an exception for mem-prec 2 and 3.

**Comment by rodykersten** [comment]
It was closed because it now reports the possible NullPointerException for mem-prec 0. For mem-prec 2 and 3, this should be fixed with #74. Almost done with that.
